jwql/website/apps/jwql/data_containers.py

```diff
@@ -882,35 +899,189 @@ def get_thumbnails_all_instruments(instruments):
+                                filenames.append(filename)
+                    else:
+                        if filters:
+                            if result['filter'] in filters[inst.lower()]:
+                                filename = result['filename'].split('.')[0]
+                                filenames.append(filename)
+                        else:
+                            filename = result['filename'].split('.')[0]
+                            filenames.append(filename)
 
         # Get list of all thumbnails
         thumbnails = glob.glob(os.path.join(THUMBNAIL_FILESYSTEM, '*', '*.thumb'))
-
-    thumbnail_list.extend(thumbnails)
+        thumbnail_list.extend(thumbnails)
 
     # Get subset of preview images that match the filenames
-    thumbnails = [os.path.basename(item) for item in thumbnail_list if
-                  os.path.basename(item).split('_integ')[0] in filenames]
+    thumbnails_subset = [os.path.basename(item) for item in thumbnail_list if
+                         os.path.basename(item).split('_integ')[0] in filenames]
 
-    return thumbnails
+    # Eliminate any duplicates
+    thumbnails_subset = list(set(thumbnails_subset))
+
+    # Determine whether or not queried anomalies are flagged
+    final_subset = []
+    for thumbnail in thumbnails_subset:
+        rootname = thumbnail.split("_")[0]+"_"+thumbnail.split("_")[1]+"_"+thumbnail.split("_")[2]+"_"+thumbnail.split("_")[3]
+        try:
+            instrument = JWST_INSTRUMENT_NAMES_SHORTHAND[thumbnail.split("_")[3][:3]]
+            thumbnail_anomalies = get_current_flagged_anomalies(rootname, instrument)
+            if thumbnail_anomalies:
+                for anomaly in anomalies[instrument.lower()]:
+                    if anomaly.lower() in thumbnail_anomalies:
+                        print(thumbnail, "contains an anomaly selected in the query")
+                        final_subset.append(thumbnail)
+        except KeyError:
+            try:
+                instrument = JWST_INSTRUMENT_NAMES_SHORTHAND[thumbnail.split("_")[2][:3]]
+                thumbnail_anomalies = get_current_flagged_anomalies(rootname, instrument)
+                if thumbnail_anomalies:
+                    for anomaly in anomalies[instrument.lower()]:
+                        if anomaly.lower() in thumbnail_anomalies:
+                            print(thumbnail, "contains an anomaly selected in the query")
+                            final_subset.append(thumbnail)
+            except KeyError:
+                print("Error with thumbnail: ", thumbnail)
+    if not final_subset:
+        print("No images matched anomaly selection")
+        final_subset = thumbnails_subset
+        if not final_subset:
+            final_subset = thumbnails[:10]
+
+    return list(set(final_subset))
+    # return thumbnails_subset  # thumbnails matching all but anomaly criteria
+
+
+def get_jwqldb_table_view_components(request):
```

Comment:
Hmm its strange that this function got added to this PR since this was Mees' work(?).  Could you remove this from this PR so that we don't get confused as to when/where it was added to the code base?

Reply:
That is strange! Sorry about that, I'm not sure what happened there. It looks like there's another version of this function on line 704. Shall I remove both versions?

Reply:
No I think that one can be kept since it is not shown as a proposed change